ts/annotations/annotations.src.ts

```diff
@@ -0,0 +1,1608 @@
+            drag?: Function;
+        }
+        interface AnnotationControlPointOptionsObject {
+            draggable?: undefined;
+            events: AnnotationControlPointEventsOptionsObject;
+            height: number;
+            index?: number;
+            positioner: AnnotationControlPointPositioner;
+            style: CSSObject;
+            symbol: string;
+            visible: boolean;
+            width: number;
+        }
+        interface AnnotationControlPointPositioner {
+            (target: AnnotationControllable): PositionObject;
+        }
+        type AnnotationDraggableValue = (''|'x'|'y'|'xy');
+        type AnnotationLabelType = AnnotationControllableLabel;
+        type AnnotationShapeType = AnnotationControllable;
+        interface AnnotationMockPointOptionsObject {
+            x: number;
+            y: number;
+            xAxis?: (number|string|Axis|null);
+            yAxis?: (number|string|Axis|null);
+        }
+        interface AnnotationPoint extends Point {
+            series: AnnotationSeries;
+        }
+        interface AnnotationSeries extends Series {
+            chart: AnnotationChart;
+            points: Array<AnnotationPoint>;
+        }
+        interface AnnotationsEventsOptions {
+            afterUpdate?: EventCallbackFunction<Annotation>;
+            add?: EventCallbackFunction<Annotation>;
+            remove?: EventCallbackFunction<Annotation>;
+        }
+        interface AnnotationsLabelOptions extends AnnotationControllableOptionsObject {
+            align: AlignValue;
+            allowOverlap: boolean;
+            backgroundColor: ColorType;
+            borderColor: ColorType;
+            borderRadius: number;
+            borderWidth: number;
+            className: string;
+            crop: boolean;
+            distance?: number;
+            format?: string;
+            formatter: FormatterCallbackFunction<Point>;
+            overflow: DataLabelsOverflowValue;
+            padding: number;
+            shadow: (boolean|ShadowOptionsObject);
+            shape: SymbolKeyValue;
+            style: CSSObject;
+            text?: string;
+            useHTML: boolean;
+            verticalAlign: VerticalAlignValue;
+            x: number;
+            y: number;
+        }
+        interface AnnotationsLabelsOptions extends AnnotationsLabelOptions {
+            point?: (string|AnnotationMockPointOptionsObject);
+        }
+        interface AnnotationsOptions extends AnnotationControllableOptionsObject {
+            controlPointOptions: AnnotationControlPointOptionsObject;
+            draggable: AnnotationDraggableValue;
+            events: AnnotationsEventsOptions;
+            id?: (number|string);
+            labelOptions: AnnotationsLabelOptions;
+            labels: Array<AnnotationsLabelsOptions>;
+            shapeOptions: AnnotationsShapeOptions;
+            shapes: Array<AnnotationsShapesOptions>;
+            typeOptions: AnnotationsTypeOptions;
```

Comment:
I think these `typeOptions` should be defined in specific files (e.g. Fibonacci, ElliottWave) etc. Every type can provide different set of options.

Reply:
As you can merge interfaces but not property typing, I kept the generic type for inheritance purpose.